ui/src/app/views/unregistered/component.spec.ts

```diff
@@ -116,11 +148,43 @@ describe('UnregisteredComponent', () => {
       requestedIdVerification: false,
     });
 
+    expectAllRegistrationSubmitted(profileStub.profile);
+  }));
+
+  it('should retry failed registration steps', fakeAsync(() => {
+    profileStub = new FailingProfileStub(2);
+    // Hacky, but unfortunately profileStub does not directly implement ProfileService.
```

Comment:
Should we move it to do so?

Reply:
Done. I'd tried this in the past and gotten stuck a few times. It's not super clean because swagger codegen does not give us an interface, just an implementation, so we need to use extension. This means we can't really enforce by compilation that we'd have a stub which implements all API methods.

That said, you will get errors if those stub methods aren't implemented, or alternatively you can pass through a mock HTTP. I think this might actually be a reasonable way forward, and I've switched the profile service stub to use this approach. If we decide we like this, we can switch the other stubs to follow this pattern later. PTAL.